tests/src/test/scala/ironrust/FullIntegrationTest.scala

```diff
@@ -503,27 +503,44 @@ class FullIntegrationTest extends DudeSuite with CancelAfterFailure {
     }
   }
 
-  "Document Encrypt Unmanaged" should {
+  "Document unmanaged encrypt/decrypt" should {
     "succeed for good data" in {
       val sdk = IronSdk.initialize(createDeviceContext)
       val data: Array[Byte] = List(1, 2, 3).map(_.toByte).toArray
       val maybeResult =
         Try(
-          sdk
-            .advanced()
+          sdk.advanced
             .documentEncryptUnmanaged(data, DocumentEncryptOpts.create(null, null, true, Array(), Array(), null))
         ).toEither
       val result = maybeResult.value
       result.getId.getId.length shouldBe 32
       result.getEncryptedDeks.isEmpty shouldBe false
     }
 
+    "roundtrip for single level transform for good data" in {
+      val sdk = IronSdk.initialize(createDeviceContext)
+      val data: Array[Byte] = List(10, 2, 3).map(_.toByte).toArray
+      val maybeResult = Try(sdk.advanced.documentEncryptUnmanaged(data, new DocumentEncryptOpts())).toEither
+      val result = maybeResult.value
+      result.getId.getId.length shouldBe 32
+
+      //Now try to decrypt
+      val maybeDecrypt =
+        Try(sdk.advanced.documentDecryptUnmanaged(result.getEncryptedData, result.getEncryptedDeks)).toEither
+      val decryptedResult = maybeDecrypt.value
+
+      decryptedResult.getId.getId.length shouldBe 32
```

Comment:
Maybe assert that `decryptedResult.getId.getId shouldBe result.getId.getId`

Reply:
Done